homeassistant/util/unit_system.py

```diff
@@ -100,12 +100,35 @@ def temperature(self: object, temperature: float, from_unit: str) -> float:
                                         from_unit, self.temperature_unit)
 
     def length(self: object, length: float, from_unit: str) -> float:
-        """Convert the given length to this unit system."""
+        """Convert the given length to this unit system return a float."""
+        converted = self.length_with_display_obj(length, from_unit)
+        return converted["value"]
+
+    def length_with_display_obj(self: object, length: float,
```

Comment:
`display_obj` is a weird suffix here. Since we are asking to convert, renaming it `length_and_unit` is more appropriate, as that is what is changing.